crates/compiler/mono/src/drop_specialization.rs

```diff
@@ -211,7 +211,20 @@ fn specialize_drops_stmt<'a, 'i>(
                     // TODO perhaps we need the union_layout later as well? if so, create a new function/map to store it.
                     environment.add_union_child(*structure, *binding, *tag_id, *index);
                     // Generated code might know the tag of the union without switching on it.
-                    // So if we unionAtIndex, we must know the tag and we can use it to specialize the drop.
+                    // So if we UnionAtIndex, we must know the tag and we can use it to specialize the drop.
+                    environment.symbol_tag.insert(*structure, *tag_id);
+                    alloc_let_with_continuation!(environment)
+                }
+                Expr::UnionFieldPtrAtIndex {
+                    structure,
+                    tag_id,
+                    union_layout: _,
+                    index,
+                } => {
+                    // TODO perhaps we need the union_layout later as well? if so, create a new function/map to store it.
+                    environment.add_union_child(*structure, *binding, *tag_id, *index);
```

Comment:
I don't think the pointer should belong as `union_child`. We don't want it used in specialization.